lib/spack/spack/test/cmd/pkg.py

```diff
@@ -236,3 +236,51 @@ def test_pkg_fails_when_not_git_repo(monkeypatch):
     monkeypatch.setattr(spack.cmd, 'spack_is_git_repo', lambda: False)
     with pytest.raises(spack.main.SpackCommandError):
         pkg('added')
+
+
+def test_pkg_source(mock_packages):
+    fake_source = pkg("source", "fake")
+
+    fake_file = spack.repo.path.filename_for_package_name("fake")
+    with open(fake_file) as f:
+        contents = f.read()
+        assert fake_source == contents
+
+
+def test_pkg_canonical_source(mock_packages):
+    source = pkg("source", "multimethod")
+    assert "@when('@2.0')" in source
+    assert "Check that multimethods work with boolean values" in source
+
+    canonical_1 = pkg("source", "--canonical", "multimethod@1.0")
+    assert "@when" not in canonical_1
```

Comment:
I get worried when I see `assert ... not in ...`, since `not in` could be true if it spit out the completely wrong info as well. Would it be too brittle if we took the output of the `spack pkg source --canonical multimethod@1.0` command as another `.txt` file and simply compared it directly to the output here?

Reply:
I think that's pretty brittle, and there are a lot of `in` tests in addition to `not in` tests here.

Reply:
Ok great!